ReactWindows/ReactNative/Views/WebView/ReactWebViewManager.cs

```diff
@@ -0,0 +1,289 @@
+        public override void ReceiveCommand(Windows.UI.Xaml.Controls.WebView view, int commandId, JArray args)
+        {
+            switch (commandId)
+            {
+                case CommandGoBack:
+                    if (view.CanGoBack) view.GoBack();
+                    break;
+
+                case CommandGoForward:
+                    if (view.CanGoForward) view.GoForward();
+                    break;
+
+                case CommandReload:
+                    view.Refresh();
+                    break;
+
+                default:
+                    throw new InvalidOperationException(
+                        $"Unsupported command '{commandId}' received by '{typeof(ReactWebViewManager)}'.");
+            }
+        }
+
+        /// <summary>
+        /// Called when view is detached from view hierarchy and allows for 
+        /// additional cleanup by the <see cref="ReactWebViewManager"/>.
+        /// </summary>
+        /// <param name="reactContext">The react context.</param>
+        /// <param name="view">The view.</param>
+        public override void OnDropViewInstance(ThemedReactContext reactContext, Windows.UI.Xaml.Controls.WebView view)
+        {
+            view.NavigationCompleted -= OnNavigationCompleted;
+            view.NavigationStarting -= OnNavigationStarting;
+            view.NavigationFailed -= OnNavigationFailed;
+        }
+
+        /// <summary>
+        /// Creates a new view instance of type <see cref="Windows.UI.Xaml.Controls.WebView"/>.
+        /// </summary>
+        /// <param name="reactContext">The react context.</param>
+        /// <returns>The view instance.</returns>
+        protected override Windows.UI.Xaml.Controls.WebView CreateViewInstance(ThemedReactContext reactContext)
+        {
+            return new Windows.UI.Xaml.Controls.WebView();
+        }
+
+        /// <summary>
+        /// Subclasses can override this method to install custom event 
+        /// emitters on the given view.
+        /// </summary>
+        /// <param name="reactContext">The react context.</param>
+        /// <param name="view">The view instance.</param>
+        protected override void AddEventEmitters(ThemedReactContext reactContext, Windows.UI.Xaml.Controls.WebView view)
+        {
+            view.NavigationCompleted += OnNavigationCompleted;
+            view.NavigationStarting += OnNavigationStarting;
+            view.NavigationFailed += OnNavigationFailed;
+        }
+
+        private async void OnNavigationCompleted(object sender, WebViewNavigationCompletedEventArgs e)
+        {
+            var webView = (Windows.UI.Xaml.Controls.WebView)sender;
+            var reactContext = webView.GetReactContext();
+            var script = default(string);
+
+            if (_injectedJS.TryGetValue(webView.GetTag(),out script) && script.Length > 0)
+            {
+                string[] args = { script };
+                await webView.InvokeScriptAsync("eval", args);
```

Comment:
Is it the correct semantics to not fire the loading finished event until after the script?